### PR TITLE
Only autolink URLs containing protocol

### PR DESCRIPTION
**Pull request by Raudius**
* Resolves: https://github.com/nextcloud/text/issues/2723
* Target version: master

### CODE REVIEW

**Review by juliushaertl — APPROVED** [review]

**Review by vinicius73 — CHANGES_REQUESTED** [review]
LGTM;
But tests should be updated.
https://github.com/nextcloud/text/runs/7501209373?check_suite_focus=true

https://github.com/nextcloud/text/blob/0b8750d158aeac8168322a9e2a60c1507d33f961/cypress/e2e/links.spec.js#L43-L55

**Comment by susnux** [comment]
I guess restricting autolink to http(s) is fine, so lgtm after fixing the tests.

File: cypress/e2e/links.spec.js [hunk 1]
```diff
@@ -45,8 +45,8 @@ describe('test link marks', function() {
 				.type('google.com')
 
 			cy.getContent()
-				.find('a[href*="google.com"]')
-				.click({ force: true })
+				.get('a[href*="google.com"]')
```
Comment (susnux) [inline cypress/e2e/links.spec.js:48]: You should stick with `find` as `get` will not only search the editor content but the whole document.
Probably something like `getContent().find(...).should('not.exist')` would work

File: cypress/e2e/links.spec.js [hunk 2]
```diff
@@ -45,8 +45,8 @@ describe('test link marks', function() {
...
+				.expect([])
 
 			cy.get('@winOpen')
```
Comment (susnux) [inline cypress/e2e/links.spec.js:51]: This part is not needed anymore as there is no link, thus no windows is opened.

**Review by susnux — CHANGES_REQUESTED** [review]

File: cypress/e2e/links.spec.js (hunk 1 above)
Comment (Raudius) [reply cypress/e2e/links.spec.js]: Thanks for the tip, I am still a cypress noobie :)

File: cypress/e2e/links.spec.js (hunk 2 above)
Comment (Raudius) [reply cypress/e2e/links.spec.js]: Removed!

**Review by vinicius73 — APPROVED** [review]

**Review by susnux — APPROVED** [review]

**Pull request merged**